Website/src/app/components/map.tsx

```diff
@@ -48,10 +49,19 @@ function Map({
 	// find the vehicle that is in focus, but only if either the vehicles, or the focus changes.
 	const vehicleInFocus = useMemo(() => vehicles.find(v => v.id == focus), [vehicles, focus]);
 
-
 	// create icons for each poi type
-	const enriched_poi_types: (POIType & {leaf_icon: L.Icon})[] = useMemo(
-		() => poi_types.map(pt => ({ ...pt, leaf_icon: L.icon({ iconUrl: pt.icon, iconSize: [45, 45] }) })),
+	const enriched_poi_types: (POIType & { leaf_icon: L.Icon })[] = useMemo(
+		() =>
+			poi_types.map(pt => {
+				const icon_src = POIIconImg[pt.icon] ?? POIIconImg[POITypeIconValues.Generic];
```

Comment:
The logic for creating the marker icon is now a little bit more complex than before. If the icon value is not one of the known ones, we will just use the "Generic" icon.